tests/test_core.py

```diff
@@ -1193,13 +1256,58 @@ def fiddle_(cls, value: int) -> None:
     def test_no_duplicate_protocols(self):
         """An Objective-C class cannot adopt a protocol more than once."""
 
+        ObjCProtocol.auto_rename = False
+
         with self.assertRaises(ValueError):
 
             class DuplicateProtocol(
                 NSObject, protocols=[NSObjectProtocol, NSObjectProtocol]
             ):
                 pass
 
+    def test_protocol_auto_rename(self):
+        """Test automatic renaming system of ObjCProtocol."""
+
+        # Check that the global option works.
+
+        ObjCProtocol.auto_rename = True
+
+        class TestProtocol(metaclass=ObjCProtocol):
+            @objc_method
+            def oldMethod(self):
+                pass
+
+        testprotocol1 = TestProtocol
+
+        class TestProtocol_2(metaclass=ObjCProtocol):
+            pass
+
+        class TestProtocol(metaclass=ObjCProtocol):  # noqa: F811
+            @objc_method
+            def newMethod(self):
+                pass
+
+        self.assertEqual(TestProtocol.name, "TestProtocol_3")
+        self.assertIsNot(testprotocol1, TestProtocol)
+
+        # Check that the per-protocol option works.
```

Comment:
As above; this should probably be split into 2 tests, with try/finally isolation, and class names that will minimise the chance of accidental overlap with other tests.